Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -25,30 +25,68 @@
 
 #include "config.h"
 #include "ScrollAnchoringController.h"
+#include <wtf/text/TextStream.h>
 
 namespace WebCore {
 
-void ScrollAnchoringController::invalidateAnchorElement()
+void ScrollAnchoringController::invalidateAnchorBox()
 {
-    m_anchorElement = nullptr;
-    m_lastPositionForAnchorElement = FloatPoint();
+    m_anchorBox = nullptr;
+    m_lastPositionForAnchorBox = FloatSize();
 }
 
-void ScrollAnchoringController::selectAnchorElement()
+FloatSize ScrollAnchoringController::computeOffset(RenderBox* candidate)
 {
-    // TODO: implement
-    // perform Anchor Element Selection algorithm on dom elements within frame view's viewable region
-    // Firefox appears to only iterate through frame tree under the current frame (unclear if
-    // looking through frame tree is sufficient)
-    // unclear to me where to call this from 
+    return candidate->location() - m_frameView.rectForFixedPositionLayout().location();
```

Comment:
`RenderBox::location()` is relative to its containing block, but you really need "absolute" (i.e. document-relative) coordinates here, or "client" (relative to viewport) coordinates. If you use `client` coordinates, check that they are relative to the `layoutViewport` (they may deviate on zooming).

Also, instead of `rectForFixedPositionLayout`, use `layoutViewport`.

Reply:
Fixed.